### PR TITLE
Add Opsgenie connector

### PR DESCRIPTION
**Pull request by huongtn**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [ #364] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: sources/opsgenie-source/src/streams/incidents.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        syncMode: SyncMode,
+        cursorField?: string[],
+        streamSlice?: Dictionary<any>,
+        streamState?: IncidentState
+    ): AsyncGenerator<Incident> {
+        const lastCreatedAt =
+            syncMode === SyncMode.INCREMENTAL
+                ? new Date(streamState?.lastCreatedAt ?? 0)
+                : undefined;
+        const buildkite = OpsGenie.instance(this.config, this.logger);
```
Comment (willmarks) [inline sources/opsgenie-source/src/streams/incidents.ts:41]: nit: `buildkite` -> `opsGenie` or something.

Comment (willmarks) [inline sources/opsgenie-source/src/streams/incidents.ts:41]: nit: `buildkite` -> `opsGenie` or something.

File: destinations/airbyte-faros-destination/src/converters/opsgenie/common.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import {Converter, parseObjectConfig, StreamContext} from '../converter';
+
+type ApplicationMapping = Record<string, {name: string; platform?: string}>;
+
+const MAX_DESCRIPTION_LENGTH = 1000;
+interface OpsGenieConfig {
+  application_mapping?: ApplicationMapping;
+  // Max length for free-form description text fields such as incident description
+  max_description_length?: number;
```
Comment (willmarks) [inline destinations/airbyte-faros-destination/src/converters/opsgenie/common.ts:11]: Do we need to add these to the destination spec?

Comment (willmarks) [inline destinations/airbyte-faros-destination/src/converters/opsgenie/common.ts:11]: Do we need to add these to the destination spec?

File: destinations/airbyte-faros-destination/src/converters/opsgenie/users.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      },
+      {
+        model: 'tms_User',
+        record: {
+          uid: user.id,
+          emailAddress: user.username,
+          name: user.fullName,
+          source,
+        },
+      },
```
Comment (willmarks) [inline destinations/airbyte-faros-destination/src/converters/opsgenie/users.ts:38]: Not sure if we want to make `tms_Users` here. @cjwooo thoughts?

File: destinations/airbyte-faros-destination/resources/spec.json [hunk 4]
```diff
@@ -514,6 +514,25 @@
...
+                  {
+                    "type": "object",
+                    "title": "Configuration",
+                    "properties": {
+                      "max_description_length": {
+                        "type": "integer",
+                        "title": "Max Description Length",
+                        "description": "Value to cut description to specific length",
+                        "default": 1000
+                      }
```
[the remark this answers is not in the document]
Comment (huongtn) [reply destinations/airbyte-faros-destination/resources/spec.json:542]: I added

File: sources/opsgenie-source/resources/spec.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{
+    "documentationUrl": "https://docs.faros.ai",
+    "connectionSpecification": {
+        "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (willmarks) [inline sources/opsgenie-source/resources/spec.json:4]: For some reason this only works on my local airbyte instance when this is `draft-07`.

Comment (huongtn) [reply sources/opsgenie-source/resources/spec.json:4]: thanks!

**Comment by cjwooo** [comment]
@huongtn Can you send me the config for source acceptance testing

**Comment by huongtn** [comment]
> @huongtn Can you send me the config for source acceptance testing

https://onetimesecret.com/secret/njffqideh3e2i08sslvry99ihvdaxg0

File: destinations/airbyte-faros-destination/src/converters/opsgenie/incidents.ts [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    const detail = status;
+    switch (status) {
+      case IncidentStatus.open:
+        return {category: IncidentStatusCategory.Investigating, detail};
+      case IncidentStatus.resolve:
+        return {category: IncidentStatusCategory.Resolved, detail};
+      case IncidentStatus.close:
+      default:
+        return {category: IncidentStatusCategory.Custom, detail};
+    }
```
Comment (willmarks) [inline destinations/airbyte-faros-destination/src/converters/opsgenie/incidents.ts:186]: I think that `resolve` should be `resolved` and `close` should be `closed`. Both of those statuses are falling into the custom default right now:
Screen Shot 2022-04-15 at 8 57 54 AM
Screen Shot 2022-04-15 at 9 00 02 AM

File: destinations/airbyte-faros-destination/src/converters/opsgenie/incidents.ts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+          ...incidentRef,
+          title: incident.message,
+          description: incident.description?.substring(0, maxDescriptionLength),
+          url: incident.links.web,
+          createdAt,
+          updatedAt,
+          acknowledgedAt,
+          resolvedAt,
+          priority: this.getPriority(incident.priority),
+          //severity: this.getSeverity(incident.severity),
```
Comment (willmarks) [inline destinations/airbyte-faros-destination/src/converters/opsgenie/incidents.ts:92]: If we don't need severity should we remove the [category mapping here](https://github.com/faros-ai/airbyte-connectors/pull/381/files#diff-6f7934fc19426d4cb3267609fa18285c88b459cadc076c7639c5a5eb87d4fac6R136-R143)?

**Comment by willmarks** [comment]
Other than my 2 recent comments, things are looking good. I think we will be good to go when these last changes are added!

**Comment by huongtn** [comment]
> 




> Other than my 2 recent comments, things are looking good. I think we will be good to go when these last changes are added!

ok let me update

File: destinations/airbyte-faros-destination/src/converters/opsgenie/incidents.ts (hunk 7 above)
Comment (huongtn) [reply destinations/airbyte-faros-destination/src/converters/opsgenie/incidents.ts:92]: ok done

**Pull request merged**